async-await.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+import 'babel/polyfill';
+import {readFile} from 'fs-promise';
+import {join} from 'path';
+
+async function main() {
+  const dir = process.argv[2];
+  let files = await readFile(join(dir, 'index.txt'));
+  let data = await* files
+    .toString()
+    .match(/^.*(?=\n)/gm)
+    .map((fileName) => join('input', fileName))
+    .map((filePath) => readFile(filePath));
```

Comment:
For an arrow function with one parameter the parens are optional. Let's drop them.

In fact, I'd prefer to map just once:

```javascript
.map(filename => readFile(join('input', filename)));
```